pkg/test/framework/components/echo/kube/instance.go

```diff
@@ -161,23 +146,215 @@ func newInstance(ctx resource.Context, originalCfg echo.Config) (out *instance,
 	return c, nil
 }
 
-func setupVM(ctx resource.Context, c *instance, cfg echo.Config) error {
+// createVMConfig sets up a Service account,
+func createVMConfig(ctx resource.Context, c *instance, cfg echo.Config) error {
```

Comment:
@stevenctl just a nit: there's a lot of code here, would it be worthwhile separating this out into another package and then breaking this down into multiple helper methods?

Reply:
Probably yeah. I'm hoping a good bit of this will be moved into application code in a follow up.